cabby/model/landmark_recognition/dataset_bert.py

```diff
@@ -0,0 +1,180 @@
+# coding=utf-8
+# Copyright 2020 Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl import logging
+from nltk.tokenize import word_tokenize
+import pandas as pd
+import torch
+from transformers import AutoTokenizer
+from typing import Dict, Tuple, Any, List
+
+from cabby.model import datasets
+
+tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")
+
+
+class EntityRecognitionSplit(torch.utils.data.Dataset):
+  """A split of the Entity Recognition dataset ."""
+
+  def __init__(self, data: pd.DataFrame):
+    # Tokenize instructions and corresponding labels.
+    basic_tokenization = [
+      basic_tokenize_and_align_labels(sent, labs)
+      for sent, labs in zip(data.instructions.tolist(), data.entity_span)
+    ]
+    self.inputs = [bert_tokenize_and_align_labels(sent, labs) for sent, labs in basic_tokenization]
+    self.sent = data.instructions.tolist()
+
+
+  def __getitem__(self, idx: int):
+    '''Supports indexing such that TextGeoDataset[i] can be used to get
+    i-th sample.
+    Arguments:
+      idx: The index for which a sample from the dataset will be returned.
+    Returns:
+      A single sample including text, the tokenization of the text and
+      the corresponding labels.
+    '''
+
+    input = {k: torch.tensor(v) for k, v in self.inputs[idx].items()}
+    input['instructions'] = self.sent[idx]
+    return input
+
+  def __len__(self):
+    return len(self.inputs)
+
+def create_dataset(
+  data_dir: str,
+  region: str,
+  s2level: int,
+) -> Tuple[EntityRecognitionSplit, EntityRecognitionSplit, EntityRecognitionSplit]:
+  '''Loads data and creates datasets and train, validate and test sets.
+  Arguments:
+    data_dir: The directory of the data.
+    region: The region of the data.
+    s2level: The s2level of the cells.
+  Returns:
+    The train, validate and test sets.
+  '''
+  rvs_dataset = datasets.RVSDataset(data_dir, s2level, region)
+
+  train_dataset = EntityRecognitionSplit(
+    rvs_dataset.train)

```

Comment:
Extraneous new lines in this block, like the other one.

Reply:
Fixed.